src/libPersistence/Retriever.cpp

```diff
@@ -95,19 +105,21 @@ bool Retriever::RetrieveBlockLink(bool trimIncompletedBlocks) {
            std::get<BlockLinkIndex::INDEX>(b);
   });
 
-  if (!blocklinks.empty()) {
-    if (m_mediator.m_ds->m_latestActiveDSBlockNum == 0) {
-      bytes latestActiveDSBlockNumVec;
-      if (!BlockStorage::GetBlockStorage().GetMetadata(
-              MetaType::LATESTACTIVEDSBLOCKNUM, latestActiveDSBlockNumVec)) {
-        LOG_GENERAL(WARNING, "Get LatestActiveDSBlockNum failed");
-        return false;
+  if (!putInIncremental) {
```

Comment:
Any reason for skipping this part is `putInIncreamental` being true?

Reply:
The operation here is useless for incremental db as it fetches the latest active block from metadata db. Actually i did not store metadata db, so it created some issues.

Reply:
I suggest to store metadata db, because once we use incremental db completely the issue this part of code trying to fix will come up again if you disabled it